### ISSUE TITLE
Are the client type annotations required?

### ISSUE THREAD

**Issue by canoon** [vemel/mypy_boto3_builder#31]
Is there a reason you didn't do something like the following to avoid specifying the service when calling .client?

```
from typing import Generic, TypeVar

T = TypeVar("T")

class S3Client():
    def put_bucket(self):
        pass


class RDSClient():
    def describe_db_instances(self):
        pass

class AWSService(Generic[T]):
    pass


S3: AWSService[S3Client] = "s3"

RDS: AWSService[RDSClient] = "rds"


def client(service: AWSService[T]) -> T:
    pass

client(S3).put_bucket()
client(RDS).describe_db_instances()
```

**Comment by vemel** [comment]
This was done and you can generate these annotations with 
```bash
mypy_boto3
````

However, there are some issues:

- VSCode does not  support literals in annotations
- PyCharm consumes tons for RAM while revealing types for literal annotations - this is the main reason why annotations are generated dynamically instead of just having static files with 200+ annotations overloads.

I did not try `Generic` approach though that you suggested, will try it this week.  If performance will not be an issue - I will add it. Thank you!

**Comment by vemel** [comment]
I am glad to announce that it has been addressed in `1.14.54.1` release. Please let me know if it works for you as wexpected.
